tests/order_modifiers_integration/Orders/Orders_Test.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace TEC\Tickets\Commerce\Order_Modifiers;
+
+use Codeception\TestCase\WPTestCase;
+use tad\Codeception\SnapshotAssertions\SnapshotAssertions;
+use TEC\Tickets\Commerce\Cart\Unmanaged_Cart;
+use TEC\Tickets\Commerce\Gateways\Stripe\Gateway as StripeGateway;
+use TEC\Tickets\Commerce\Order_Modifiers\Models\Fee;
+use TEC\Tickets\Commerce\Order_Modifiers\Models\Order_Modifier_Meta;
+use TEC\Tickets\Commerce\Order_Modifiers\Models\Order_Modifier_Relationships as Relationships_Model;
+use TEC\Tickets\Commerce\Order_Modifiers\Repositories\Order_Modifiers_Meta as Repository;
+use TEC\Tickets\Commerce\Order_Modifiers\Repositories\Order_Modifier_Relationship as Relationship_Repository;
+use TEC\Tickets\Commerce\Order_Modifiers\Values\Float_Value;
+use Tribe\Tickets\Test\Commerce\Attendee_Maker;
+use Tribe\Tickets\Test\Commerce\TicketsCommerce\Order_Maker;
+use Tribe\Tickets\Test\Commerce\TicketsCommerce\Ticket_Maker;
+use Tribe\Tickets\Test\Traits\Reservations_Maker;
+use Tribe\Tickets\Test\Traits\With_Tickets_Commerce;
+use WP_Post;
+
+class Orders_Test extends WPTestCase {
+	use Ticket_Maker;
+	use Attendee_Maker;
+	use With_Tickets_Commerce;
+	use Reservations_Maker;
+	use SnapshotAssertions;
+	use Order_Maker;
+
+	/** @var ?Unmanaged_Cart */
+	protected ?Unmanaged_Cart $cart;
+
+	/** @var ?Repository */
+	protected ?Repository $repository;
+
+	protected ?Relationship_Repository $relationship_repository;
+
+	/**
+	 * @before
+	 */
+	public function set_up() {
+		$this->cart = $this->cart ?? new Unmanaged_Cart();
+		$this->cart->clear();
+		$this->repository              = $this->repository ?? new Repository();
+		$this->relationship_repository = $this->relationship_repository ?? new Relationship_Repository();
+	}
+
+	/**
+	 * @test
+	 * @dataProvider order_totals_data_provider
+	 * Ensures the order totals are calculated correctly for tickets and fees.
+	 */
+	public function it_calculates_order_totals_with_various_inputs(
+		Float_Value $ticket_price,
+		Float_Value $fee_raw_amount,
+		string $fee_application,
+		Float_Value $expected_total,
+		Float_Value $expected_subtotal
+	): void {
+		// Create an event post.
+		$event_id = self::factory()->post->create(
+			[
+				'post_title' => 'The Event',
+			]
+		);
+
+		// Create a fee with the specified raw amount.
+		$fee = $this->create_fee( [ 'raw_amount' => $fee_raw_amount ] );
+		$this->set_fee_application( $fee, $fee_application );
+
+		// Associate the fee with the event.
+		$this->relationship_repository->insert(
+			new Relationships_Model(
+				[
+					'modifier_id' => $fee->id,
+					'post_id'     => $event_id,
+					'post_type'   => 'post',
+				]
+			)
+		);
```

Comment:
I need to debug this section. I am attaching the fee to the post, but it isn't taking into consideration when calculating the subtotal/total. I think the issue is in the test though.